src/nodes/Geometry3D/IndexedFaceSet.js

```diff
@@ -932,111 +943,18 @@ x3dom.registerNodeType(
                                             this._mesh._texCoords[0].push(texCoords[t2].z);
                                         }
                                     }
-
```

Comment:
I do not have the stomach to think this through. So I would err on the side of caution, and separate this out for now. Somebody else should review this as well. Since this affects most IFSs and most scenes have those, a good test would be see if all web3d and x3dom examples with IFSs still work.